### PR TITLE
[DO NOT MERGE without @cvializ] Format dollaramount as ("cents" -> cents)

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@camhashemi @leogrim

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentRequest.scala [hunk 1]
```diff
@@ -31,8 +31,8 @@ object AccountStateResponse {
   implicit val writes = (
     (__ \ 'users).write[Int] and
     (__ \ 'billingDate).write[DateTime] and
-    (__ \ 'balance).write(DollarAmount.dollarStringFormat) and
-    (__ \ 'charge).write(DollarAmount.dollarStringFormat) and
+    (__ \ 'balance).write(DollarAmount.formatAsCents) and
+    (__ \ 'charge).write(DollarAmount.formatAsCents) and
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentRequest.scala:35]: Is @cvializ ready for this? I wouldn't merge until we are

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PaidAccount.scala [hunk 2]
```diff
@@ -22,22 +24,17 @@ case class DollarAmount(cents: Int) extends Ordered[DollarAmount] {
...
-  val dollarStringFormat = new Format[DollarAmount] {
-    def writes(o: DollarAmount) = JsString(o.toDollarString)
-
-    // this is a fragile reads, shouldn't be used anywhere but tests until improved or fully-spec'd
-    def reads(json: JsValue): JsResult[DollarAmount] = json.validate[String].map { str =>
-      val centsString = str.drop(str.indexOf('$') + 1).replace(".", "")
-      DollarAmount(centsString.toInt)
-    }
-  }
+  val formatAsCents: Format[DollarAmount] = (__ \ 'cents).format[Int].inmap(DollarAmount.cents, _.toCents)
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PaidAccount.scala:37]: Why not just straight to numeric?

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentRequest.scala (hunk 1 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentRequest.scala:35]: :+1: I'll let him drive on this one.

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PaidAccount.scala (hunk 2 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PaidAccount.scala:37]: More extensible, and more semantically clear (so it's easier for us to debug issues when they come up). I don't have super strong opinions about this, but I prefer it for the same reason that I prefer `varchar` to `boolean` in the db (even when we only have two states).

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentRequest.scala (hunk 1 above)
Comment (cvializ) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentRequest.scala:35]: Let it henceforth be known that we're sending an object with

```
{
 currencyUnit: integer_value
}
```

not a string, not a raw number.

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PaidAccount.scala (hunk 2 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PaidAccount.scala:37]: Alright. As long as we don't get carried away :)

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentRequest.scala (hunk 1 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentRequest.scala:35]: His excellency has spoken.

**Comment by cvializ** [comment]
Should work, look out below : O

**Pull request merged**